lib/control/validation/token.js

```diff
@@ -3,50 +3,64 @@
 const HTTP = require('http');
 const bad = require('../validation/badRequest');
 
-function send(body, ami, vault, callback) {
+function send(body, ami, path, vault, callback) {
+  return new Promise((resolve, reject) => {
+    const request = new Buffer(JSON.stringify(body, null, 2));
 
-  const request = new Buffer(JSON.stringify(body, null, 2));
-
-  const req = HTTP.request({
-    hostname: vault.hostname,
-    port: vault.port,
-    method: 'POST',
-    path: '/v1/auth/token/create',
-    headers: {
-      'X-Vault-Token': vault.token,
-      'content-length': request.length
-    }
-  });
+    const req = HTTP.request({
+      hostname: vault.hostname,
+      port: vault.port,
+      method: 'POST',
+      path,
+      headers: {
+        'X-Vault-Token': vault.token,
+        'content-length': request.length
+      }
+    });
 
-  const chunks = [];
+    const chunks = [];
 
-  req.on('response', function(res) {
-    res.on('end', function ended(chunk) {
-      if (chunk) { chunks.push(chunk); }
-      try {
-        const response = JSON.parse(Buffer.concat(chunks).toString('utf8'));
+    req.on('response', function (res) {
+      res.on('end', function ended(chunk) {
+        if (chunk) {
+          chunks.push(chunk);
+        }
+        try {
+          const response = JSON.parse(Buffer.concat(chunks)
+              .toString('utf8'));
 
-        callback(null, response);
-      } catch(e) {
-        callback(e);
-      }
+          resolve(response);
+        } catch (e) {
+          reject(e);
+        }
+      });
+      res.on('data', (chunk) => chunks.push(chunk));
     });
-    res.on('data', (chunk) => chunks.push(chunk));
-  });
 
-  req.on('error', callback);
+    req.on('error', (err) => reject(err));
 
-  req.write(request);
-  req.end();
+    req.write(request);
+    req.end();
+  });
 }
 
-exports.send = send;
+function getTokenTiming(accessor, vault) {
+  return send({}, '', `/v1/auth/token/lookup-accessor/${accessor}`, vault).then((resp) => {
+    const creation_time = resp.data.creation_time * 1000;
+    const explicit_max_ttl = resp.data.explicit_max_ttl * 1000;
```

Comment:
If `explicit_max_ttl` isn't set on the token, Vault will come back with 0 for that field. In that case, we need to read `/sys/mounts/auth/token/tune` to get the `max_lease_ttl` value for tokens.